### PR TITLE
llvm: begin the journey of independence from llvm

### PR DESCRIPTION
**Pull request by jacobly0**
These changes don't accomplish anything concrete quite yet, but they do vastly reduce the amount of information flowing across the ABI boundary from LLVM to the Zig compiler.

### CODE REVIEW

File: src/main.zig [hunk 1]
```diff
@@ -439,6 +439,8 @@ const usage_build_generic =
     \\  -fno-unwind-tables        Never produce unwind table entries
     \\  -fLLVM                    Force using LLVM as the codegen backend
     \\  -fno-LLVM                 Prevent using LLVM as the codegen backend
+    \\  -flibLLVM                 Force using LLVM shared library apias the codegen backend
```
Comment (matu3ba) [inline src/main.zig:442]: s/apias/as

File: src/main.zig [hunk 2]
```diff
@@ -439,6 +439,8 @@ const usage_build_generic =
...
+    \\  -fno-libLLVM              Prevent using LLVM as the codegen backend
```
Comment (matu3ba) [inline src/main.zig:443]: `s/Prevent using LLVM as the codegen backend/Prevent using LLVM shared library as the codegen backend`.

File: src/main.zig (hunk 1 above)
Comment (andrewrk) [reply src/main.zig:442]: Would you mind making the new ones lower case? `-fLLVM` and `-fno-LLVM` are the only non-lower case flags - apologies for that - and I plan to rename them as well.

Comment (andrewrk) [reply src/main.zig:442]: haha, fair! feel free to leave it as-is, I'll change them all at once just before the release.

**Comment by andrewrk** [comment]
Nice work getting this to a mergeable point. Changes look great.

**Comment by andrewrk** [comment]
Perf data point:

```
Benchmark 1 (3 runs): before/bin/zig build-exe --stack 33554432 /home/andy/Downloads/zig/src/main.zig --cache-dir /home/andy/Downloads/zig/zig-cache --global-cache-dir /home/andy/.cache/zig --name zig --mod build_options::/home/andy/Downloads/zig/zig-cache/c/ecc8358ae8fa382c4a1d0b2cb9404f45/options.zig --deps build_options
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          57.3s  ± 1.73s     55.3s  … 58.5s           0 ( 0%)        0%
  peak_rss           3.47GB ±  277KB    3.47GB … 3.47GB          0 ( 0%)        0%
  cpu_cycles          230G  ± 1.27G      229G  …  232G           0 ( 0%)        0%
  instructions        333G  ± 67.3M      333G  …  333G           0 ( 0%)        0%
  cache_references   13.3G  ± 93.2M     13.3G  … 13.4G           0 ( 0%)        0%
  cache_misses       1.04G  ± 18.9M     1.02G  … 1.06G           0 ( 0%)        0%
  branch_misses      1.60G  ± 1.26M     1.60G  … 1.60G           0 ( 0%)        0%
Benchmark 2 (3 runs): after/bin/zig build-exe --stack 33554432 /home/andy/Downloads/zig/src/main.zig --cache-dir /home/andy/Downloads/zig/zig-cache --global-cache-dir /home/andy/.cache/zig --name zig --mod build_options::/home/andy/Downloads/zig/zig-cache/c/ecc8358ae8fa382c4a1d0b2cb9404f45/options.zig --deps build_options
  measurement          mean ± σ            min … max           outliers         delta
  wall_time          57.5s  ± 21.8ms    57.5s  … 57.5s           0 ( 0%)          +  0.3% ±  4.8%
  peak_rss           3.81GB ± 30.8MB    3.79GB … 3.84GB          0 ( 0%)        💩+  9.6% ±  1.4%
  cpu_cycles          237G  ± 1.19G      236G  …  238G           0 ( 0%)        💩+  2.8% ±  1.2%
  instructions        342G  ± 1.85G      341G  …  344G           0 ( 0%)        💩+  2.9% ±  0.9%
  cache_references   13.6G  ± 65.4M     13.6G  … 13.7G           0 ( 0%)          +  2.3% ±  1.4%
  cache_misses       1.08G  ± 4.94M     1.08G  … 1.09G           0 ( 0%)        💩+  4.5% ±  3.0%
  branch_misses      1.63G  ± 11.4M     1.63G  … 1.65G           0 ( 0%)        💩+  2.2% ±  1.1%
```

**Pull request merged**